test/cpp/api/modules.cpp

```diff
@@ -803,18 +803,27 @@ TEST_F(ModulesTest, Bilinear) {
 }
 
 TEST_F(ModulesTest, Fold) {
-  Fold model(FoldOptions({4, 5}, {2, 2}));
-  auto x = torch::randn({1, 3 * 2 * 2, 12}, torch::requires_grad());
-  auto y = model(x);
-  torch::Tensor s = y.sum();
+  {
+    Fold model(FoldOptions({4, 5}, {2, 2}));
+    auto x = torch::randn({1, 3 * 2 * 2, 12}, torch::requires_grad());
+    auto y = model(x);
+    torch::Tensor s = y.sum();
 
-  s.backward();
-  ASSERT_EQ(y.ndimension(), 4);
-  ASSERT_EQ(s.ndimension(), 0);
-  ASSERT_EQ(y.size(0), 1);
-  ASSERT_EQ(y.size(1), 3);
-  ASSERT_EQ(y.size(2), 4);
-  ASSERT_EQ(y.size(3), 5);
+    s.backward();
+    ASSERT_EQ(y.ndimension(), 4);
+    ASSERT_EQ(s.ndimension(), 0);
+    ASSERT_EQ(y.size(0), 1);
+    ASSERT_EQ(y.size(1), 3);
+    ASSERT_EQ(y.size(2), 4);
+    ASSERT_EQ(y.size(3), 5);
```

Comment:
It would be awesome to have an expected value test for `y` as well :D Thanks so much for your help!